### PR TITLE
Fix typecheck

### PR DESCRIPTION
**Pull request by Atticus1806**
Attempt on fixing the typecheck problems in #10 . @mmz33 can you maybe test if this fixes the problems also for you? At least in my setup it fixed raised warnings and failed tests. 
Also raised this here, maybe there is a better way https://github.com/agronholm/typeguard/issues/358.

### CODE REVIEW

**Comment by Atticus1806** [comment]
As an additional discussion point: Do we want to fix the typeguard version in the requirements? Maybe to 4.0.0? This is what I tested my code with right now.

File: i6_models/config.py [hunk 1]
```diff
@@ -29,7 +29,18 @@ class ModelConfiguration:
     def _validate_types(self) -> None:
         for field in fields(type(self)):
             try:
-                typeguard.check_type(getattr(self, field.name), field.type)
+                if isinstance(field.type, str):
```
Comment (albertz) [inline i6_models/config.py:32]: You should add some comment. Maybe referring to the issue you should open on the typeguard repo about how to handle this.

File: i6_models/config.py [hunk 2]
```diff
@@ -30,6 +30,9 @@ def _validate_types(self) -> None:
         for field in fields(type(self)):
             try:
                 if isinstance(field.type, str):
+                    # there field.type most of the time is a str object which needs to be resolved to the type itself
+                    # at the moment is does not look like there is typeguard support for it, thus manual fixes
+                    # c.f. https://github.com/agronholm/typeguard/issues/358
```
Comment (albertz) [inline i6_models/config.py:35]:
```suggestion
                    # Later Python versions use forward references,
                    # and then field.type is a str object.
                    # This needs to be resolved to the type itself.
                    # At the moment is does not look like there is typeguard support for it, thus manual fixes.
                    # C.f. https://github.com/agronholm/typeguard/issues/358
```

**Comment by mmz33** [comment]
I tested this and it fixes the issue described in #10.

**Review by albertz — APPROVED** [review]

**Review by mmz33 — APPROVED** [review]

**Pull request merged**